api/v1alpha1/virtualmachineimage_conversion_test.go

```diff
@@ -43,6 +59,36 @@ func TestVirtualMachineImageConversion(t *testing.T) {
 		g.Expect(spoke.Spec.ProviderRef.Kind).To(Equal("ImageProvider"))
 		g.Expect(spoke.Spec.ProviderRef.Name).To(Equal("my-image"))
 		g.Expect(spoke.Spec.ProviderRef.Namespace).To(Equal("my-namespace"))
+
+		g.Expect(spoke.Annotations).To(HaveLen(3))
+		g.Expect(spoke.Annotations).To(HaveKeyWithValue(sysAnnotationKey("foo"), "foo-val"))
+		g.Expect(spoke.Annotations).To(HaveKeyWithValue(sysAnnotationKey("bar"), "bar-val"))
+	})
+
+	t.Run("VirtualMachineImage spoke-hub", func(t *testing.T) {
+		nextVerCVMI := &nextver.ClusterVirtualMachineImage{}
+		spoke := &v1alpha1.ClusterVirtualMachineImage{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "foo",
+				Annotations: map[string]string{
+					sysAnnotationKey("foo"): "foo-val",
+					sysAnnotationKey("bar"): "bar-val",
+					"fizz":                  "buzz",
+				},
+			},
+		}
+
+		g.Expect(spoke.ConvertTo(nextVerCVMI)).To(Succeed())
+		g.Expect(nextVerCVMI.Annotations).To(HaveLen(1))
+		g.Expect(nextVerCVMI.Status).NotTo(BeNil())
+
+		props := nextVerCVMI.Status.VMwareSystemProperties
+		g.Expect(props).To(HaveLen(2))
+		g.Expect([]string{
```

Comment:
I wrote it this way since the ordering of the Key/Value pair is random because of the backing map. Anyway I will change it to assert the key and value pair.